src/pages/sales-order/Tabs/SalesOrderForm.js

```diff
@@ -0,0 +1,1364 @@
+        overheadId: '',
+        vatAmount: 0,
+        mdAmount: 0,
+        upo: 0,
+        extendedPrice: 0,
+        priceType: 1,
+        applyVat: false,
+        notes: ''
+      }
+    ]
+  })
+
+  const invalidate = useInvalidate({
+    endpointId: SaleRepository.SalesOrder.page
+  })
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      date: yup.string().required(),
+      currencyId: yup.string().required(),
+      clientId: yup.string().required(),
+      items: yup.array().of(
+        yup.object({
+          sku: yup.string().required(),
+          itemName: yup.string().required(),
+          qty: yup.number().required().min(1)
+        })
+      )
+    }),
+    onSubmit: async obj => {
+      try {
+        const copy = { ...obj }
+        delete copy.items
+        copy.date = formatDateToApi(copy.date)
+        copy.dueDate = formatDateToApi(copy.dueDate)
+
+        if (!obj.rateCalcMethod) delete copy.rateCalcMethod
+
+        if (copy.serializedAddress) {
+          const addressData = {
+            clientId: copy.clientId,
+            address: address
+          }
+
+          const addressRes = await postRequest({
+            extension: SaleRepository.Address.set,
+            record: JSON.stringify(addressData)
+          })
+          copy.shipToAddressId = addressRes.recordId
+        }
+
+        const updatedRows = formik.values.items.map((itemDetails, index) => {
+          const { physicalProperty, ...rest } = itemDetails
+
+          return {
+            ...rest,
+            seqNo: index + 1,
+            siteId: obj.siteId,
+            applyVat: obj.isVattable
+          }
+        })
+
+        const itemsGridData = {
+          header: copy,
+          items: updatedRows
+        }
+
+        const soRes = await postRequest({
+          extension: SaleRepository.SalesOrder.set2,
+          record: JSON.stringify(itemsGridData)
+        })
+
+        const actionMessage = editMode ? platformLabels.Edited : platformLabels.Added
+        toast.success(actionMessage)
+        await refetchForm(soRes.recordId)
+        invalidate()
+      } catch (error) {}
+    }
+  })
+
+  const editMode = !!formik.values.recordId
+  const isClosed = formik.values.wip === 2
+
+  const columns = [
+    {
+      component: 'resourcelookup',
+      label: labels.sku,
+      name: 'sku',
+      props: {
+        endpointId: InventoryRepository.Item.snapshot,
+        parameters: '_categoryId=0&_msId=0&_startAt=0&_size=1000',
+        displayField: 'sku',
+        valueField: 'recordId',
+        mapping: [
+          { from: 'recordId', to: 'itemId' },
+          { from: 'sku', to: 'sku' },
+          { from: 'name', to: 'itemName' }
+        ],
+        columnsInDropDown: [
+          { key: 'sku', value: 'SKU' },
+          { key: 'name', value: 'Item Name' }
+        ],
+        displayFieldWidth: 5
```

Comment:
isn't there a prop to send for required?

Reply:
what the purpose of it? already it's added to validation schema